### PR TITLE
Fix double logging + reduced log level for streams

### PR DESCRIPTION
**Pull request by thomheymann**
- This PR fixes an issue where prematurely closed sockets would cause duplicate log lines
- This PR also reduces the log level of prematurely closed sockets from WARN to INFO (clients closing sockets is expected behaviour)

Fixes  fastify/fastify-static#116

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -387,12 +387,20 @@ function onSendEnd (reply, payload) {
...
 
   eos(payload, { readable: true, writable: false }, function (err) {
     sourceOpen = false
     if (err != null) {
       if (res.headersSent) {
-        reply.log.warn({ err }, 'response terminated with an error with headers already sent')
+        if (!errorLogged) {
+          errorLogged = true
+          if (err.code === 'ERR_STREAM_PREMATURE_CLOSE') {
+            reply.log.info({ res }, 'stream closed prematurely')
```
Comment (jsumners) [inline lib/reply.js:399]: Do we really need to log this? The server shouldn't really care, right? Maybe log at `debug` or `trace`? I'm not sure.

Comment (thomheymann) [reply lib/reply.js:399]: Sending a normal response is currently logged as 'request complete' (INFO).
Closing the stream prematurely is the equivalent I think.
If we don't log that you would have requests that have a log line for when the request came but nothing else and thus no way of knowing what happened to the request.

Comment (jsumners) [reply lib/reply.js:399]: In that case, this should honor the request logging disablement feature.

**Comment by mcollina** [comment]
Good work @thomheymann! +1 from me once @jsumners comment is resolved.

**Comment by Eomm** [comment]
I think this is related to https://github.com/fastify/fastify-static/issues/116

File: lib/reply.js (hunk 1 above)
Comment (thomheymann) [reply lib/reply.js:399]: I can't find reference/docs for that feature. Can you point me to it?

Comment (mcollina) [reply lib/reply.js:399]: https://www.fastify.io/docs/latest/Server/#disablerequestlogging

**Comment by thomheymann** [comment]
@Eomm Yeh, while that `fastify-static` issue is not a bug specifically, the current behaviour is misleading so wanted to reduce the severity

**Comment by thomheymann** [comment]
@jsumners Good shout, I've added a condition to skip logging if request logging is disabled. Please re-check

**Comment by Eomm** [comment]
@thomheymann Would you mind to backported to `v2`?

**Comment by thomheymann** [comment]
> @thomheymann Would you mind to backported to `v2`?

Depends how you handle your versioning but I'd kind of consider this a breaking change. If anyone has their monitoring configured to look for certain WARN logs then this will break that.

**Pull request merged**